mitmproxy/addons/tlsconfig.py

```diff
@@ -583,23 +586,46 @@ def _warn_seclevel_missing(self, side: Literal["client", "server"]) -> None:
                 f"for insecure TLS versions to work."
             )
 
+    @property
+    def crl_path(self) -> str:
+        if not self.certstore:
+            return ""
+
+        return "/" + str(self.certstore.default_ca.serial) + ".crl"
+
     def get_cert(self, conn_context: context.Context) -> certs.CertStoreEntry:
         """
         This function determines the Common Name (CN), Subject Alternative Names (SANs) and Organization Name
         our certificate should have and then fetches a matching cert from the certstore.
         """
         altnames: list[x509.GeneralName] = []
         organization: str | None = None
+        crl_distribution_points: list[str] = []
 
         # Use upstream certificate if available.
         if ctx.options.upstream_cert and conn_context.server.certificate_list:
-            upstream_cert = conn_context.server.certificate_list[0]
+            upstream_cert: certs.Cert = conn_context.server.certificate_list[0]
             if upstream_cert.cn:
                 altnames.append(_ip_or_dns_name(upstream_cert.cn))
             altnames.extend(upstream_cert.altnames)
             if upstream_cert.organization:
                 organization = upstream_cert.organization
 
+            # Replace original URL path with the CA cert serial number, which acts as a magic token
+            for url in upstream_cert.crl_distribution_points:
+                urlParts = list(parse(url))
+                urlParts[3] = self.crl_path
+
+                # I hope I am just doing something stupid and that this is not the intended way to actually go about doing this
+                crl_distribution_points.append(
+                    unparse(
+                        urlParts[0].decode("ascii"),
+                        urlParts[1].decode("idna"),
+                        urlParts[2],
+                        urlParts[3],
+                    )
+                )
```

Comment:
Nit:
```suggestion
                scheme, host, port, _ = parse(url)
                url = unparse(scheme, host, port, self.crl_path)
                crl_distribution_points.append(url)
```

(+ the list[str] -> str change)

Reply:
Hmmm, the issue with that code is, it produces a cert with this type of url
```
X509v3 CRL Distribution Points: 
  Full Name:
    URI:b'http'://b'crl3.digicert.com'/mitmproxy-132645061682444739888094978381816565991132842371.crl
```
Which is why I had to do that weird way, does x509 not mind this mixing of `b'`/binary strings or...? As normal certs seem to just be a string without any binary strings in them

Reply:
Ughs - I didn't realize that `url.parse` is super old terrible code. It looks like it returns `scheme, host, port, path` as `tuple[bytes, bytes, int, bytes]`, but `unparse` then expects `str` arguments. 🤦 

Let me fix that in a separate PR in the coming days.

Reply:
Alrighty then! Ty!

Reply:
https://github.com/mitmproxy/mitmproxy/pull/7624 should have you covered. The URI should of course be
```
http://crl3.digicert.com/mitmproxy-132645061682444739888094978381816565991132842371.crl
```
and not contain any other byte markers. :)

Reply:
I am confused on what the new code should be, looking at the changes, the current code I have seems to be the correct way still? (As it checks if scheme is string and if so, prints all as a string? (and as parse returns bytes for scheme and host, I still need to decode them back into strings from bytes?))

Reply:
Forgive me for doing so but it has been almost 2 weeks, so I am writing a message to "bump" the thread